### PR TITLE
Simulation api implementation and old migration used

### PR DESCRIPTION
**Pull request by feroj21**

### CODE REVIEW

**Comment by feroj21** [comment]
[Api Simulations page](https://trello.com/c/1TClABsd/136-api-simulations-page)

File: app/Http/Controllers/Student/PageController.php [hunk 1]
```diff
@@ -43,15 +44,43 @@ class PageController extends Controller {
...
+                'completed' => $simulationsCompletedTexts,
+            ],
+            'tests' =>  [
+                'done' => $doneTakes,
+                'active' => $accessibleTests,
+            ]
+        ]);
+    }
+
+    private function _getPopups(Request $request) {
```
Comment (LCD344) [inline app/Http/Controllers/Student/PageController.php:83]: we don't need the underscore here - its ok without

File: app/Providers/RouteServiceProvider.php [hunk 2]
```diff
@@ -38,17 +39,26 @@ public function boot()
...
-                ->group(base_path('routes/api.php'));
+            $this->registerApiRoutes();
 
             Route::middleware('web')
                 ->namespace($this->namespace)
                 ->group(base_path('routes/web.php'));
         });
     }
 
+    public function registerApiRoutes() {
```
Comment (LCD344) [inline app/Providers/RouteServiceProvider.php:50]: we will need to add a different one of these for every api file?

Can we do this like we do with the web? so it is automatic?

````php
foreach (File::allFiles(__DIR__ . "/web") as $routeFile) {
	require $routeFile;
}
````

File: database/migrations/2017_06_09_143047_create_failed_jobs_table.php [hunk 3]
```diff
@@ -31,6 +30,6 @@ public function up()
      */
     public function down()
     {
-        Schema::dropIfExists('failed_jobs');
+        Schema::drop('failed_jobs');
```
Comment (LCD344) [inline database/migrations/2017_06_09_143047_create_failed_jobs_table.php:33]: is there a reason that some have Schema::drop and some Schema::dropIfExists?

**Pull request merged**